main.js

```diff
@@ -160,4 +160,116 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+function countTVtosell(televisions) {
+  let totalCount = 0;
+  for (const television of televisions) {
+    const tosellofthistype = television.originalStock - television.sold;
+    totalCount = totalCount + tosellofthistype;
+  }
+  return totalCount;
+}
+const totalCount = countTVtosell(inventory);
+
+const numbertoSell = document.getElementById("numbertoSell");
+numbertoSell.textContent = totalCount;
+
+
+const specTypes = inventory.map((inventory) => {
+  return inventory.brand + " type " + inventory.type;
+})
+//console.log(specTypes);
+
+
+const TypeSoldout = inventory.filter((inventory) => {
+  return inventory.sold == inventory.originalStock;
+})
+//console.log(TypeSoldout);
+
+
+const OptAmbi = inventory.filter((inventory) => {
+  return inventory.options.ambiLight === inventory.options.ambiLight;
+})
+//console.log(OptAmbi);
+
+const TVPriceSort = inventory.sort((TVA, TVB) => {
+  return TVA.price - TVB.price;
+});
+//console.log(TVPriceSort)
+
+
+function countRE(televisions) {
+  let totalCountRE = 0;
+  for (const television of televisions) {
+    const tosoldofthistype = television.originalStock * television.price;
+    totalCountRE = totalCountRE + tosoldofthistype;
+  }
+  return totalCountRE;
+}
+const totalCountRE = countRE(inventory);
+
+const RE = document.getElementById("RE");
+RE.textContent = "€" +totalCountRE;
+
+
+function countRT(televisions) {
+  let totalCountRT = 0;
+  for (const television of televisions) {
+    const tosoldyetofthistype = television.sold * television.price;
+    totalCountRT = totalCountRT + tosoldyetofthistype;
+  }
+  return totalCountRT;
+}
+const totalCountRT = countRT(inventory);
+
+const RT = document.getElementById("RT");
+RT.textContent = "€" +totalCountRT;
+
+//-onderstaande functie krijg ik niet gewerkt!!!!
+
+function displayProduct(product) {
+  const itemTVs = document.createElement("li");
+  itemTVs.setAttribute("class", "product");
+  const itemTV = product.type;
+  itemTV.textContent = itemTV;
+const list = document.getelementById("product");
+  list.appendChild(TVelement);
+}
+
+for (let index = 0; index < inventory.length; index++) {
+  const product = inventory[index];
+  displayProduct(product);
+}
+// bovenstaande opdracht met functie displayProduct geeft geen tv.types weer op de pagina
+// -- de verwijzing gaat ergens verkeerd, maar ik weet niet waar!!
+
+
+//Opdracht5
+
+const TVObject = inventory.map((inventory) => {
+  return inventory.brand + " | " + inventory.name + " | " + inventory.type + "| price €" + inventory.price + ",--  |" + inventory.availableSizes + "  inches --> to CM " + (inventory.availableSizes = inventory.availableSizes * 2.54);
```

Comment:
Met `inventory.availableSizes = inventory.availableSizes * 2.54` overschrijf je de waarde. `availableSizes` is ook een array en dus een size op zichzelf, maar een array met mogelijke sizes. Het had duidelijker en makkelijker geweest als je een functie had gemaakt die dat voor je afhandelt:

```
function getScreenSizes(television) {
  const tvSizes = television.availableSizes.map(size => {
      const sizeRoundedCm = Math.round(size * 2.54);
      return size + " inch (" + sizeRoundedCm + " cm)";
  };

  return tvSizes.join(" | ");
}
```

Reply:
De display van de producten naam, type, prijs heb ik voor elkaar gekregen.